plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryArrowBufferAllocator.java

```diff
@@ -45,8 +48,8 @@ public BufferAllocator newChildAllocator(BigQuerySplit split)
         return rootAllocator.newChildAllocator(
                 split.streamName(),
                 new RetryingAllocationListener(split.streamName(), stats),
-                split.dataSize().orElse(0),
-                maximumAllocation);
+                0, // no reservation
```

Comment:
nit: We extract a constant `private static final long NO_RESERVATION = 0;` instead.